test/framework/src/main/java/org/elasticsearch/test/AbstractBuilderTestCase.java

```diff
@@ -449,6 +449,7 @@ private static class ServiceHolder implements Closeable {
             List<NamedWriteableRegistry.Entry> entries = new ArrayList<>();
             entries.addAll(IndicesModule.getNamedWriteables());
             entries.addAll(searchModule.getNamedWriteables());
+            pluginsService.forEach(plugin -> entries.addAll(plugin.getNamedWriteables()));
```

Comment:
Are there named writeables that aren't queries or anything? I don't understand the need for this?

Reply:
I originally added this because early on in test development, the inference call used named writeables that weren't registered. Since we are properly simulating the inference call in the test now, the unregistered named writeables aren't used and this can be removed. Good catch!

Reply:
... And I needed to add this back when I extended the serialization tests in https://github.com/elastic/elasticsearch/pull/108483/commits/27dff4b8cbed2e2d0d20ba95ad1b5773af8d660e. `rewriteQuery` now checks that the query can be serialized in between when it is rewritten on the coordinator & data nodes. For the semantic query, this means serializing `InferenceResults`, which in turn requires registering the named writeables for that class.